### PR TITLE
hardening/754_add_tests_for_post_subscriptions

### PR DESCRIPTION
**Pull request by pcoello25**
* Partially fix issue #754 
  * Update CHANGES_NEXT_RELEASE is not necessary.
    * POST `/v1/subscriptions?ngsi_version=<ngsi_version>` post a valid subscription
* 100% unit tests passed:
```
Running com.telefonica.iot.cygnus.management.ManagementInterfaceTest
[ManagementInterface] -------------------- 'POST method posts a valid subscription (ngsi_version = 1)'.
[ManagementInterface] --------------------  OK  - Valid subscription
[ManagementInterface] -------------------- 'POST method posts a valid subscription (ngsi_version = 2)'.
[ManagementInterface] --------------------  OK  - Valid subscription
[ManagementInterface] -------------------- 'POST method post a subscription with missing fields (ngsi_version = 1)'.
[ManagementInterface] --------------------  OK  - Subscription has missing fields
[ManagementInterface] -------------------- 'POST method post a subscription with missing fields (ngsi_version = 2)'.
[ManagementInterface] --------------------  OK  - Subscription has missing fields
[ManagementInterface] -------------------- 'POST method doesn't find a subscription (ngsi_version = 1)'.
[ManagementInterface] --------------------  OK  - Subscription not found
[ManagementInterface] -------------------- 'POST method doesn't find a subscription (ngsi_version = 2)'.
[ManagementInterface] --------------------  OK  - Subscription not found
[ManagementInterface] -------------------- 'POST method post a subscription with empty fields (ngsi_version = 1)'.
[ManagementInterface] --------------------  OK  - Subscription has empty fields
[ManagementInterface] -------------------- 'POST method post a subscription with empty fields (ngsi_version = 2)'.
[ManagementInterface] --------------------  OK  - Subscription has empty fields
[ManagementInterface] -------------------- Testing ManagementInterface.handle
Tests run: 9, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 1.407 sec
```
* Assignee: @frbattid

### CODE REVIEW

File: cygnus-common/src/main/java/com/telefonica/iot/cygnus/management/ManagementInterface.java [hunk 1]
```diff
@@ -1561,6 +1559,9 @@ private String setCorrelator (HttpServletRequest request) {
...
         return corrId;
     } // setCorrelator
+    
+    protected void setOrionBackend(OrionBackendImpl orionBackend) {
+        this.orionBackend = orionBackend;
+    } // setOrionBackend
 
-} // ManagementInterface
-
+} // ManagementInterface
```
Comment (frbattid) [inline cygnus-common/src/main/java/com/telefonica/iot/cygnus/management/ManagementInterface.java]: White line is required

Comment (pcoello25) [inline cygnus-common/src/main/java/com/telefonica/iot/cygnus/management/ManagementInterface.java]: Fixed in 2cb01a0

File: cygnus-common/src/main/java/com/telefonica/iot/cygnus/management/ManagementInterface.java [hunk 2]
```diff
@@ -1561,6 +1559,9 @@ private String setCorrelator (HttpServletRequest request) {
         MDC.put(CommonConstants.LOG4J_TRANS, transId);
         return corrId;
     } // setCorrelator
+    
+    protected void setOrionBackend(OrionBackendImpl orionBackend) {
```
Comment (frbattid) [inline cygnus-common/src/main/java/com/telefonica/iot/cygnus/management/ManagementInterface.java]: Most probably your IDE is asking for Javadoc comments, isn't it?

File: cygnus-common/src/test/java/com/telefonica/iot/cygnus/management/ManagementInterfaceTest.java [hunk 3]
```diff
@@ -31,25 +36,78 @@
...
+            httpStatus = sc;
+            super.sendError(sc);
+        }
+
+        @Override
+        public void sendError(int sc, String msg) throws IOException {
+            httpStatus = sc;
+            super.sendError(sc, msg);
+        }
+
```
Comment (frbattid) [inline cygnus-common/src/test/java/com/telefonica/iot/cygnus/management/ManagementInterfaceTest.java]: Extra white line

File: cygnus-common/src/test/java/com/telefonica/iot/cygnus/management/ManagementInterfaceTest.java [hunk 4]
```diff
@@ -31,25 +36,78 @@
...
+        public void setStatus(int sc) {
+            httpStatus = sc;
+            super.setStatus(sc);
+        }
+
+        public int getStatus() {
+            return httpStatus;
+        }
+
+    }
```
Comment (frbattid) [inline cygnus-common/src/test/java/com/telefonica/iot/cygnus/management/ManagementInterfaceTest.java]: In general, comments when closing `}` are welcome :)

File: cygnus-common/src/test/java/com/telefonica/iot/cygnus/management/ManagementInterfaceTest.java [hunk 5]
```diff
@@ -84,6 +205,225 @@ public void testHandle() {
...
+        System.out.println(getTestTraceHead("[ManagementInterface]") + "- 'POST method posts a valid subscription (ngsi_version = 1)'.");
+        StatusExposingServletResponse responseWrapper = new StatusExposingServletResponse(response);
+        ManagementInterface managementInterface = new ManagementInterface(new File(""), null, null, null, 8081, 8082);
+        managementInterface.setOrionBackend(orionBackend);
+             
+        try {
+            managementInterface.handlePostSubscription(mockRequestV1, responseWrapper);
+        } catch (Exception x) {
+            System.out.println("There was some problem when handling the POST subscription");
+            throw x;
```
Comment (frbattid) [inline cygnus-common/src/test/java/com/telefonica/iot/cygnus/management/ManagementInterfaceTest.java]: Call me freak, but I would prefer `e` for an exception variable :)

Comment (pcoello25) [inline cygnus-common/src/test/java/com/telefonica/iot/cygnus/management/ManagementInterfaceTest.java]: Fixed in 7db24d8

File: cygnus-common/src/test/java/com/telefonica/iot/cygnus/management/ManagementInterfaceTest.java (hunk 4 above)
Comment (pcoello25) [inline cygnus-common/src/test/java/com/telefonica/iot/cygnus/management/ManagementInterfaceTest.java]: Fixed in 7db24d8

File: cygnus-common/src/test/java/com/telefonica/iot/cygnus/management/ManagementInterfaceTest.java (hunk 3 above)
Comment (pcoello25) [inline cygnus-common/src/test/java/com/telefonica/iot/cygnus/management/ManagementInterfaceTest.java]: Fixed in 7db24d8

File: cygnus-common/src/main/java/com/telefonica/iot/cygnus/management/ManagementInterface.java (hunk 2 above)
Comment (pcoello25) [inline cygnus-common/src/main/java/com/telefonica/iot/cygnus/management/ManagementInterface.java]: Fixed in 7db24d8

**Comment by frbattid** [comment]
LGTM

**Pull request merged**